### PR TITLE
[jit] builtin-sum 

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: torch/csrc/jit/script/compiler.cpp [hunk 1]
```diff
@@ -428,6 +428,7 @@ struct Environment {
           {"abs", std::make_shared<BuiltinFunction>(prim::abs, at::nullopt)},
           {"all", std::make_shared<BuiltinFunction>(aten::all, at::nullopt)},
           {"list", std::make_shared<BuiltinFunction>(aten::list, at::nullopt)},
+          {"sum", std::make_shared<BuiltinFunction>(aten::sum, at::nullopt)},
```
Comment (zou3519) [inline torch/csrc/jit/script/compiler.cpp]: Does this work for something like `[1, 2]` (a non-tensor argument?)

**Comment by zou3519** [comment]
@jamesr66a could you take a look at this?

File: test/test_jit.py [hunk 2]
```diff
@@ -6524,6 +6524,19 @@ def int2():
         self.checkScript(int1, ())
         self.checkScript(int2, ())
 
+    def test_number_sum(self):
+        def int1():
```
Comment (jamesr66a) [inline test/test_jit.py]: `int1` and `float1` need to take an argument

**Comment by jamesr66a** [comment]
Hey sorry for the delay. This is something that we want but we would want it to support all the built-in types and not just tensor, for example how we do it with ops in register_prim_ops: https://github.com/pytorch/pytorch/blob/7201a2e854fdf0f08f98aad3c5f9696883de5eb2/torch/csrc/jit/register_prim_ops.cpp#L1489

**Comment by michaelarg** [comment]
Hi @jamesr66a, I've been a bit held up with commitments lately but I will get back onto pytorch development in the coming weeks. Looking forward to the future collaboration.

**Pull request closed without merging**